impeller/renderer/backend/vulkan/test/mock_vulkan.cc

```diff
@@ -16,16 +18,34 @@ struct MockCommandBuffer {
   std::shared_ptr<std::vector<std::string>> called_functions_;
 };
 
-struct MockDevice {
-  MockDevice() : called_functions_(new std::vector<std::string>()) {}
+class MockDevice final {
+ public:
+  explicit MockDevice() : called_functions_(new std::vector<std::string>()) {}
+
   MockCommandBuffer* NewCommandBuffer() {
-    std::unique_ptr<MockCommandBuffer> buffer =
-        std::make_unique<MockCommandBuffer>(called_functions_);
+    auto buffer = std::make_unique<MockCommandBuffer>(called_functions_);
     MockCommandBuffer* result = buffer.get();
+    Lock lock(command_buffers_mutex_);
     command_buffers_.emplace_back(std::move(buffer));
     return result;
   }
+
+  std::shared_ptr<std::vector<std::string>>& GetCalledFunctions() {
```

Comment:
```suggestion
  std::shared_ptr<const std::vector<std::string>>& GetCalledFunctions() {
```

Now that someone has to use `was_called` it's worth making this return a const pointer so someone doesn't accidentally do the wrong thing.